src/components/Messages/Message.tsx

```diff
@@ -20,13 +22,23 @@ const Message: FC = () => {
   const messageState = useMessageStore((state) => state);
   const { conversations } = messageState;
   const selectedConversation = conversations.get(address);
-  const { messages } = useGetMessages(selectedConversation);
+  const [endTime, setEndTime] = useState<Date>();
+  const { messages } = useGetMessages(selectedConversation, endTime);
   const { sendMessage } = useSendMessage(selectedConversation);
 
   const onConversationSelected = (address: string) => {
     push(address ? `/messages/${address}` : '/messages/');
   };
 
+  const fetchNextMessages = () => {
```

Comment:
This should be memoized, since it gets passed to a component.

Reply:
Do you want me to use, useCallback here?

because memo would only make sense if it had an input, and it does not have one

Reply:
Yes, exactly

Reply:
Okay this is done